daemon/create.go

```diff
@@ -151,5 +153,9 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 	if err != nil {
 		return nil, err
 	}
+
+	if (driverName != "" && v.DriverName() != driverName) || (driverName == "" && v.DriverName() != volume.DefaultDriverName) {
```

Comment:
@azurezk this check should probably be inside `volumes.Create()` otherwise the volume is already created. See @coolljt0725's #16278 PR that has a better logic IMO.